### ISSUE TITLE
Usage of "m/" in derivation path

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-bitcoin#2671]
In #2451 we removed the ability to parse derivation paths that include the leading 'm'. While testing the rc1 I attempted to upgrade `bitcoind` which includes an RPC call to `getaddressinfo` which returns the following JSON. As can be seen the derivation path includes the 'm'. This means our current release candidate is unable to parse JSON returned by Bitcoin Core, surely that is a bug?

TRACE][bitcoincore_rpc]: JSON-RPC response for getaddressinfo:
```json
{
    "address":"n1fg6HFS7TV7ETSvCWD68k6ZNEsXYWg88U",
    "scriptPubKey":"76a914dd08bbd7d4526a708aa9346a50918c411a860d7388ac",
    "ismine":true,
    "solvable":true,
    "desc":"pkh([5041878d/0'/0'/6']033c9bfc9acc37f679ac7099fc0eda080efca7b454a35818f8fd9d11a79245c725)#5p87z2p4",
    "iswatchonly":false,
    "isscript":false,
    "iswitness":false,
    "pubkey":"033c9bfc9acc37f679ac7099fc0eda080efca7b454a35818f8fd9d11a79245c725",
    "iscompressed":true,
    "ischange":false,
    "timestamp":1712535496,
    "hdkeypath":"m/0'/0'/6'",
    "hdseedid":"0510f7177cb0e3f37c22b27d222befe4e797bcc5",
    "hdmasterfingerprint":"5041878d",
    "labels":[""]
}

```

**Comment by apoelstra** [comment]
cc @josibake we removed the ability to parse `m` as part of a literal BIP32 path in your https://github.com/rust-bitcoin/rust-bitcoin/pull/2451 and associated issue because it was causing "confusion" for students.

And yet we see that Bitcoin Core allows parsing the `m` and it also unconditionally serializes the `m`. Should we revert your change?

https://github.com/bitcoin/bitcoin/blob/master/src/util/bip32.cpp

**Comment by josibake** [comment]
Based on the argument in https://github.com/rust-bitcoin/rust-bitcoin/issues/2449 , I’d still say treating `m` as a constant is incorrect (based on a strict reading of the BIP). However, if this is so widely done that it would be too cumbersome to change, we could revert the change here and amend the BIP.

I can also open a PR in core to see what the appetite for changing it there is

**Comment by apoelstra** [comment]
> I can also open a PR in core to see what the appetite for changing it there is

For sure, could you do this? Would be good to see some discussion from other Core people.

> we could revert the change here and amend the BIP.

Let's open a PR in Core and give it a couple days, then yes, I think we have to revert the change here (or at least, soften our parsing code to *allow* `m`) to match existing Core logic. And depending on the outcome of the Core discussion probably we'll need to amend the BIP in some way. (But I wouldn't start there, given the unclear maintainership status of the BIP repo right now.)

**Comment by josibake** [comment]
@apoelstra @tcharding worth mentioning, the descriptor field _does_ have the correctly serialized path (without `m`). The `hdkeypath` is a leftover from legacy wallets and afaict descriptors always serialize the path without `m`.

I'll still open a PR in core but just wanted to mention this in case it helps unblock you guys.

**Comment by tcharding** [comment]
To help resolve this I opened:
- https://github.com/rust-bitcoin/rust-bitcoin/pull/2674 to fix the release
- https://github.com/rust-bitcoin/rust-bitcoin/issues/2673 to track that we should dig a little harder

**Comment by tcharding** [comment]
Fixed by #2677
